force-app/src/classes/MethodSpy.cls

```diff
@@ -16,17 +16,16 @@
  */
 @IsTest
 global class MethodSpy {
-  global String methodName { get; private set; }
-  global CallLog callLog { get; private set; }
-  private List<ParameterizedMethodSpyCall> parameterizedMethodCalls = new List<ParameterizedMethodSpyCall>();
-  private Object returnValue;
-  private Boolean configuredGlobalReturn = false;
-  private Boolean configuredGlobalThrow = false;
-  private Exception exceptionToThrow;
+  global final String methodName { get; private set; }
+  global final CallLog callLog { get; private set; }
+  private final List<ParameterizedMethodSpyCall> methodCalls;
+  private final BehaviorManagement behaviorManager;
```

Comment:
suggestion: rename either to `defaultBehaviorManager` or `globalBehaviorManager`